### PR TITLE
Added SBT contracts

### PR DESCRIPTION
**Pull request by aritkulova**
- [ ] Since this PR suggests a **bug fix**, the tests have been added and the coverage is 100%.
- [x] Since this PR introduces a **new feature**, the update has been discussed in an Issue or with the team.
- [ ] This PR is just a **minor change**, like a typo fix.

+ abstract SBT
+ ISBT
+ OwnableSBT
+ SBTMock

### CODE REVIEW

File: contracts/tokens/extensions/WhitelistedSBT.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+contract WhitelistedSBT is SBT {
+    mapping(address => bool) internal _whitelisted;
+
+    function __WhitelistedSBT_init(
+        string calldata name_,
+        string calldata symbol_,
+        string calldata baseTokenURI_,
+        address[] calldata addresses_
+    ) public onlyInitializing {
```
Comment (Hrom131) [inline contracts/tokens/extensions/WhitelistedSBT.sol:15]: Change to internal

File: contracts/tokens/extensions/WhitelistedSBT.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        __SBT_init(name_, symbol_, baseTokenURI_);
+        _addToWhitelist(addresses_);
+    }
+
+    function isWhitelisted(address address_) public view returns (bool) {
+        return _whitelisted[address_];
+    }
+
+    function _mint(address to_, uint256 tokenId_) internal override {
```
Comment (Hrom131) [inline contracts/tokens/extensions/WhitelistedSBT.sol:24]: Add virtual

File: contracts/tokens/SBT.sol [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    function name() public view virtual override returns (string memory) {
+        return _name;
+    }
+
+    function symbol() public view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    function baseURI() public view virtual returns (string memory) {
```
Comment (Hrom131) [inline contracts/tokens/SBT.sol:45]:
```suggestion
    function baseURI() public view virtual override returns (string memory) {
```

File: contracts/tokens/SBT.sol [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    function name() public view virtual override returns (string memory) {
+        return _name;
+    }
+
+    function symbol() public view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    function baseURI() public view virtual override returns (string memory) {
```
Comment (Arvolear) [inline contracts/tokens/SBT.sol:45]: Let's add an internal function to set this

File: contracts/tokens/SBT.sol [hunk 5]
```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {ISBT} from "../interfaces/tokens/ISBT.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
```
Comment (Arvolear) [inline contracts/tokens/SBT.sol:7]:
```suggestion
import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";

import {ISBT} from "../interfaces/tokens/ISBT.sol";

```

File: contracts/interfaces/tokens/ISBT.sol [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/**
+ * @notice The SBT module
+ */
+interface ISBT {
+    event Minted(address to, uint256 tokenId);
```
Comment (RuslanProgrammer) [inline contracts/interfaces/tokens/ISBT.sol:8]: Add comments for events

File: contracts/tokens/SBT.sol [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+     * - If individual token URI is set, it gets returned.
+     * - Otherwise if base URI is set, the concatenation of base URI and token URI gets returned.
+     * - Otherwise `""` gets returned
+     *
+     * @param tokenId_ the token to get the URI of
+     * @return the URI of the token
+     */
+    function tokenURI(uint256 tokenId_) public view virtual override returns (string memory) {
+        if (bytes(_tokenURIs[tokenId_]).length != 0) {
+            return _tokenURIs[tokenId_];
```
Comment (RuslanProgrammer) [inline contracts/tokens/SBT.sol:122]: Should we cache the value of `_tokenURIs[tokenId_]`?

File: contracts/tokens/SBT.sol [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+     * @notice The function to get the token URI.
+     *
+     * - If individual token URI is set, it gets returned.
+     * - Otherwise if base URI is set, the concatenation of base URI and token URI gets returned.
+     * - Otherwise `""` gets returned
+     *
+     * @param tokenId_ the token to get the URI of
+     * @return the URI of the token
+     */
+    function tokenURI(uint256 tokenId_) public view virtual override returns (string memory) {
```
Comment (RuslanProgrammer) [inline contracts/tokens/SBT.sol:120]: Should we process a branch when the `tokenId_` does not exist?

File: test/token/SBT.test.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+const truffleAssert = require("truffle-assertions");
+
+const SBTMock = artifacts.require("SBTMock");
+
+describe("SBT", () => {
+  const name = "testName";
+  const symbol = "TS";
+
+  let FIRST;
+  let SECOND;
```
Comment (RuslanProgrammer) [inline test/token/SBT.test.js:14]: Unused variable

File: contracts/interfaces/tokens/ISBT.sol (hunk 6 above)
Comment (Arvolear) [reply contracts/interfaces/tokens/ISBT.sol:8]: Will add it in the next releases. We should probably refactor the comments

File: contracts/tokens/SBT.sol (hunk 8 above)
Comment (Arvolear) [reply contracts/tokens/SBT.sol:120]: Reverting view functions is usually not good

**Review by RuslanProgrammer — APPROVED** [review]
Nice

**Pull request merged**